include/networkit/networkit_export.h

```diff
@@ -0,0 +1,48 @@
+
+#ifndef NETWORKIT_EXPORT_H
+#define NETWORKIT_EXPORT_H
+
+#ifdef NETWORKIT_STATIC_DEFINE
+#  define NETWORKIT_EXPORT
+#  define NETWORKIT_NO_EXPORT
+#else

```

Comment:
Why is this case needed? I think the entire file needs some comments to explain the purpose of each macro.